File: 02_Object_Oriented_Programming/02_Class_Relationships/04_Composition/java/Room.java
/**
 * Composition — Room class.
 *
 * Key concept: A House is COMPOSED OF Rooms.
 * Rooms are created and owned by the House (strong ownership).
 * If the House is destroyed, its Rooms go with it.
 */
public class Room {

    private String name;
    private double areaSqFt;

    public Room(String name, double areaSqFt) {
        this.name = name;
        this.areaSqFt = areaSqFt;
    }

    public String getName() {
        return name;
    }

    public double getAreaSqFt() {
        return areaSqFt;
    }

    @Override
    public String toString() {
        return name + " (" + areaSqFt + " sq ft)";
    }
}

File: 02_Object_Oriented_Programming/02_Class_Relationships/04_Composition/javascript/room.js
/**
 * Composition — Room class in JavaScript.
 */

class Room {
  #name;
  #areaSqFt;

  constructor(name, areaSqFt) {
    this.#name = name;
    this.#areaSqFt = areaSqFt;
  }

  get name() {
    return this.#name;
  }

  get areaSqFt() {
    return this.#areaSqFt;
  }

  toString() {
    return `${this.#name} (${this.#areaSqFt} sq ft)`;
  }
}

module.exports = Room;
